chain/x/zoracle/internal/keeper/request.go

```diff
@@ -24,16 +24,53 @@ func (k Keeper) GetRequest(ctx sdk.Context, id uint64) (types.Request, sdk.Error
 	return request, nil
 }
 
+func (k Keeper) AddSubmitValidator(ctx sdk.Context, id int64, validator sdk.ValAddress) sdk.Error {
+	request, err := k.GetRequest(ctx, id)
+	if err != nil {
+		return err
+	}
+	for _, submittedValidator := range request.SubmittedValidatorList {
+		if validator.Equals(submittedValidator) {
+			return types.ErrDuplicateValidator(types.DefaultCodespace)
+		}
+	}
+	found := false
+	for _, validValidator := range request.ValidatorList {
+		if validator.Equals(validValidator) {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return types.ErrInvalidValidator(types.DefaultCodespace)
+	}
+	request.SubmittedValidatorList = append(request.SubmittedValidatorList, validator)
+	k.SetRequest(ctx, id, request)
+	return nil
+}
+
+func (k Keeper) SetResolve(ctx sdk.Context, id int64, isResolved bool) sdk.Error {
```

Comment:
Feels really bad to load the whole struct just to set one boolean value.... But I think this is the standard. So ok.